pkg/insightstore/filestore.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightstore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pipe-cd/pipe/pkg/filestore"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+type Store struct {
+	filestore filestore.Store
+}
+
+func NewInsightFileStore(fs filestore.Store) Store {
```

Comment:
nit: `NewStore`
Because the package name is already meaning the `insight`. So the caller will call this function by: `insighstore.NewStore()`